src/v/kafka/server/materialized_partition.h

```diff
@@ -26,11 +26,11 @@ class materialized_partition final : public kafka::partition_proxy::impl {
     }
 
     model::offset high_watermark() const final {
-        return _log.offsets().dirty_offset;
+        return _log.offsets().dirty_offset + model::offset(1);
```

Comment:
@mmaslankaprv @rystsov does this look correct to you all?

Reply:
one thing it should return 0 if offset is negative, see `raft::next_offset`

Reply:
Nice catch! I think this can allow me to remove this hack i had in place in my ducktape tests: https://github.com/vectorizedio/redpanda/pull/1444/commits/ccb462bda17db78a7acd400e020bdde0a523b124

My guess is my kafka client was using a large negative value as the start offset in its fetch request due to it being returned from a previous list offsets request.